torch/onnx/_internal/io_adapter.py

```diff
@@ -611,9 +611,13 @@ def apply(
         ordered_params = tuple(
             model.state_dict[name] for name in model.graph_signature.parameters  # type: ignore[union-attr,index]
         )
-        ordered_buffers = tuple(
-            model.state_dict[name] for name in model.graph_signature.buffers  # type: ignore[union-attr,index]
-        )
+        non_persistent_buffers = set(model.graph_signature.non_persistent_buffers)  # type: ignore[union-attr]
+        ordered_buffers = []
+        for name in model.graph_signature.buffers:  # type: ignore[union-attr]
+            if name in non_persistent_buffers:
+                ordered_buffers.append(model.constants[name])  # type: ignore[union-attr]
+            else:
+                ordered_buffers.append(model.state_dict[name])  # type: ignore[union-attr,index]
```

Comment:
Hi @suo I noticed in ExportedProgram that although we are lifting `constant_tensors`, and `constants`, they are saved individually, not in state_dict. Are they going to be merged in one dict? As from graph execution perspective, backends would need to get these inputs from different places (state_dict, constant_tensors, constants). Thanks!

Reply:
No, they will not be merged. The state_dict contains parameters/buffers, non-persistent buffers are not saved at all in eager pytorch and are reinitialized each time. For export we can't do that, so we save non-persistent buffers to the constants dict.